src/billing/ConsolidatedSpendReport.tsx

```diff
@@ -458,7 +463,7 @@ export const ConsolidatedSpendReport = (props: ConsolidatedSpendReportProps): Re
                         roles: ['Owner'],
                         status: 'Ready',
                       }}
-                      key={workspace.workspaceId}
+                      key={workspace.workspaceId ?? `${workspace.namespace}-${workspace.name}`}
```

Comment:
Use `/` as a delimiter instead of `-`. Slashes are not allowed in namespaces or names, so it's a safe delimiter. Using a dash here would result in the same key for `namespace=foo, name=bar-baz` and `namespace=foo-bar, name=baz`.